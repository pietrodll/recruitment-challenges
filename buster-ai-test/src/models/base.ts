import { Pool, PoolClient, QueryConfig, QueryResult } from 'pg';
import { PoolManager } from '../services/pool';

export type WithId<T> = T & { id: string };

export abstract class BaseModel<T, U> {
    private pool: Pool;

    protected abstract table: string;

    protected abstract tableStructure: Record<keyof T, keyof U>;

    constructor() {
        this.pool = PoolManager.getInstance();
    }

    protected abstract mapToJson(dbItem: WithId<U>): WithId<T>;

    protected async runQuery(query: QueryConfig): Promise<WithId<T>[]> {
        const res = await this.pool.query(query);

        return res.rows.map(this.mapToJson);
    }

    protected async runRawQuery<R = any>(
        query: QueryConfig
    ): Promise<QueryResult<R>> {
        return this.pool.query(query);
    }

    protected getClient(): Promise<PoolClient> {
        return this.pool.connect();
    }

    protected async runTransaction<R>(
        query: QueryConfig
    ): Promise<QueryResult<R>> {
        const client = await this.getClient();

        try {
            await client.query('BEGIN');

            const res = await client.query<R>(query);

            await client.query('COMMIT');

            return res;
        } catch (e) {
            await client.query('ROLLBACK');

            throw e;
        } finally {
            client.release();
        }
    }

    public async getById(id: string): Promise<WithId<T>> {
        const query = {
            name: `get-${this.table}-by-id`,
            text: `SELECT * FROM ${this.table} WHERE id = $1 LIMIT 1`,
            values: [id],
        };

        const results = await this.runQuery(query);
        return results[0];
    }

    public async create(item: T): Promise<string> {
        const keys = Object.keys(this.tableStructure) as (keyof T)[];
        const tableKeys = keys.map(key => this.tableStructure[key]).join(', ');

        const query = {
            name: `create-${this.table}`,
            text: `INSERT INTO ${this.table} (${tableKeys}) VALUES(${keys
                .map((_, i) => `$${i + 1}`)
                .join(', ')}) RETURNING id`,
            values: keys.map(key => item[key]),
        };

        const res = await this.runTransaction<{ id: string }>(query);
        return res.rows[0].id;
    }
}
