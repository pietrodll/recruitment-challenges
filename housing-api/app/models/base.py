"""This module contains the abstract classes Entity and Model to be used as base for all the entities and models"""

from abc import ABCMeta, abstractmethod, abstractstaticmethod
from bson import ObjectId
from pymongo import ReturnDocument
from ..utils.various import SingletonMeta
from ..utils.errors import NotFoundError
from ..db import DBClient


class Entity(metaclass=ABCMeta):
    """Class representing an entity"""

    @abstractstaticmethod
    def from_json(data: dict):
        """Creates an instance of the class from a JSON dictionary. Includes fields validation"""

    @abstractmethod
    def to_json(self):
        """Returns a valid Flask response"""

    @abstractmethod
    def update_from_json(self, data: dict):
        """Updates the entity from a dictionary"""


class Model(metaclass=SingletonMeta):
    """Class representing a model in the database"""

    collection_name = None

    def __init__(self) -> None:
        self.db = DBClient()
        self.collection = self.db.db[self.collection_name]

    @abstractmethod
    def to_entity(self, record: dict):
        """Converts a database record to an `Entity` object"""

    @abstractmethod
    def from_entity(self, entity):
        """Converts an entity to a dictionary to be inserted in the database"""

    def get_by_id(self, id: str):
        """Gets an entity by ID"""

        record = self.collection.find_one(ObjectId(id))

        if record is None:
            raise NotFoundError(f'Cannot find entity with id {id}')

        return self.to_entity(record)

    def insert(self, entity):
        rec = self.from_entity(entity)

        result = self.collection.insert_one(rec)
        return result

    def replace(self, entity):
        updated = self.collection.find_one_and_replace(
            {'_id': ObjectId(entity.id)},
            self.from_entity(entity),
            return_document=ReturnDocument.AFTER
        )

        return self.to_entity(updated)

    def list(self):
        records = self.collection.find()

        if records is None:
            return []

        return list(map(self.to_entity, records))
