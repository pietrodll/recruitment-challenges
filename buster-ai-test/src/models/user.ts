import { CREDIT_DECREMENT, INIT_CREDIT } from '../constants';
import { createForbiddenError } from '../utils/errors';
import { BaseModel, WithId } from './base';

interface UserInit {
    email: string;
    pwdHash: string;
}

interface DBUser {
    email: string;
    pwd_hash: string;
    credit: number;
}

export class User {
    public email: string;
    public pwdHash: string;
    public credit: number;

    constructor(init: UserInit & Partial<User>) {
        this.email = init.email;
        this.pwdHash = init.pwdHash;
        this.credit =
            typeof init.credit == 'number' ? init.credit : INIT_CREDIT;
    }

    public static fromDB(dbItem: DBUser): User {
        return new User({
            ...dbItem,
            pwdHash: dbItem.pwd_hash,
        });
    }
}

export class UserModel extends BaseModel<User, DBUser> {
    protected mapToJson(dbItem: WithId<DBUser>): WithId<User> {
        return {
            id: dbItem.id,
            ...User.fromDB(dbItem),
        };
    }

    protected table = 'users';

    protected tableStructure: Record<keyof User, keyof DBUser> = {
        email: 'email',
        pwdHash: 'pwd_hash',
        credit: 'credit',
    };

    public async getByEmail(email: string): Promise<WithId<User>> {
        const query = {
            name: 'get-user-by-email',
            text: `SELECT * FROM ${this.table} WHERE email = $1`,
            values: [email],
        };

        const [res] = await this.runQuery(query);
        return res;
    }

    public async useCredit(id: string): Promise<number> {
        const user = await this.getById(id);

        if (user.credit < CREDIT_DECREMENT) {
            throw createForbiddenError('Not enough credit');
        }

        const query = {
            name: 'decrement-user-credit',
            text: `UPDATE ${this.table} SET credit = credit - $1 WHERE id = $2 AND credit >= $1 RETURNING credit`,
            values: [CREDIT_DECREMENT, id],
        };

        const res = await this.runTransaction<{ credit: number }>(query);
        return res.rows[0].credit;
    }

    public async addCredit(id: string, value: number): Promise<number> {
        const query = {
            name: 'increment-user-credit',
            text: `UPDATE ${this.table} SET credit = credit + $1 WHERE id = $2 RETURNING credit`,
            values: [value, id],
        };

        const res = await this.runTransaction<{ credit: number }>(query);
        return res.rows[0].credit;
    }
}

export const userModel = new UserModel();
