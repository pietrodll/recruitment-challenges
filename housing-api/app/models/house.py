"""This module contains the House and HouseModel classes"""

import re
from bson import ObjectId
from flask import jsonify
from ..utils.various import SingletonMeta
from ..utils.errors import APIError
from ..utils.validation import validate_input
from .base import Entity, Model


def rooms_validator(rooms):
    if len(rooms) == 0:
        return False

    for room in rooms:
        if 'name' not in room or 'size' not in room:
            return False

        if type(room['name']) != str or type(room['size']) != int:
            return False

    return True


HOUSE_VALIDATION = {
    'name': {
        'required': True,
        'type': str
    },
    'address': {
        'required': True,
        'type': str
    },
    'postal_code': {
        'required': True,
        'type': str,
        'pattern': re.compile(r'^\d{5}$')
    },
    'city': {
        'required': True,
        'type': str
    },
    'rooms': {
        'required': True,
        'type': list,
        'custom_validator': rooms_validator
    }
}


class House(Entity):
    def __init__(self, id, created_by, name, address, postal_code, city, rooms) -> None:
        super().__init__()

        self.id = id
        self.created_by = created_by
        self.name = name
        self.address = address
        self.postal_code = postal_code
        self.city = city
        self.rooms = rooms

    @staticmethod
    def format_rooms(raw_rooms):
        rooms = []

        for raw in raw_rooms:
            rooms.append({
                'name': raw['name'],
                'size': raw['size']
            })

        return rooms

    @staticmethod
    def from_json(data: dict):
        """Creates a `House` instance from a JSON dictionary"""

        validate_input(HOUSE_VALIDATION, data)

        return House(
            None,
            None,
            data['name'],
            data['address'],
            data['postal_code'],
            data['city'],
            House.format_rooms(data['rooms'])
        )

    def to_json(self):
        """Creates a valid Flask response"""

        return {
            'id': self.id,
            'created_by': self.created_by,
            'name': self.name,
            'address': self.address,
            'postal_code': self.postal_code,
            'city': self.city,
            'rooms': self.rooms
        }

    def update_from_json(self, data: dict):
        """Updates the user according to a dictionary"""

        validate_input(HOUSE_VALIDATION, data, check_required=False)

        if 'name' in data:
            self.name = data['name']

        if 'address' in data:
            self.address = data['address']

        if 'postal_code' in data:
            self.postal_code = data['postal_code']

        if 'city' in data:
            self.city = data['city']

        if 'rooms' in data:
            self.rooms = House.format_rooms(data['rooms'])


class HouseModel(Model, metaclass=SingletonMeta):
    collection_name = 'houses'

    def to_entity(self, record: dict):
        return House(
            str(record['_id']),
            record['created_by'],
            record['name'],
            record['address'],
            record['postal_code'],
            record['city'],
            record['rooms']
        )

    def from_entity(self, entity):
        if entity.created_by is None:
            raise APIError('Cannot add to database a house without owner ID')

        rec = {
            'created_by': entity.created_by,
            'name': entity.name,
            'address': entity.address,
            'postal_code': entity.postal_code,
            'city': entity.city,
            'rooms': entity.rooms
        }

        if entity.id is not None:
            rec['_id'] = ObjectId(entity.id)

        return rec

    def search_by_city(self, city: str):
        records = self.collection.find({
            'city': {'$regex': city, '$options': 'i'}
        })

        if records is None:
            return []

        return list(map(self.to_entity, records))

    def get_by_owner(self, owner_id: str):
        """Returns the list of the houses of an owner"""

        records = self.collection.find({'created_by': owner_id})

        if records is None:
            return []

        return list(map(self.to_entity, records))
