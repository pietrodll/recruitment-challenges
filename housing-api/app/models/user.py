"""This module contains the User and the UserModel classes"""

import re
from datetime import datetime
from bson import ObjectId
from flask import jsonify
from ..utils.various import SingletonMeta
from ..utils.validation import validate_input
from .base import Entity, Model


USER_VALIDATION = {
    'email': {
        'required': True,
        'type': str
    },
    'first_name': {
        'required': True,
        'type': str
    },
    'last_name': {
        'required': True,
        'type': str
    },
    'birth_date': {
        'required': True,
        'type': str,
        'pattern': re.compile(r'^\d{4}-\d{2}-\d{2}$')
    }
}


class User(Entity):
    def __init__(self, id: str, email: str, password_hash: str, first_name: str, last_name: str, birth_date: datetime) -> None:
        super().__init__()

        self.id = id
        self.email = email
        self.password_hash = password_hash
        self.first_name = first_name
        self.last_name = last_name
        self.birth_date = birth_date

    @staticmethod
    def from_json(data: dict):
        """Creates a `User` instance from a JSON dictionary"""

        validate_input(USER_VALIDATION, data)

        return User(
            None,
            data['email'],
            None,
            data['first_name'],
            data['last_name'],
            datetime.fromisoformat(data['birth_date'])
        )

    def to_json(self):
        """Creates a valid Flask response"""

        return jsonify({
            'id': self.id,
            'email': self.email,
            'first_name': self.first_name,
            'last_name': self.last_name,
            'birth_date': self.birth_date.isoformat()
        })

    def update_from_json(self, data: dict):
        """Updates the user according to a dictionary"""

        validate_input(USER_VALIDATION, data, check_required=False)

        if 'last_name' in data:
            self.last_name = data['last_name']

        if 'first_name' in data:
            self.first_name = data['first_name']

        if 'birth_date' in data:
            self.birth_date = datetime.fromisoformat(data['birth_date'])


class UserModel(Model, metaclass=SingletonMeta):
    collection_name = 'users'

    def to_entity(self, record: dict):
        return User(
            str(record['_id']),
            record['email'],
            record['password_hash'],
            record['first_name'],
            record['last_name'],
            datetime.fromisoformat(record['birth_date']),
        )

    def from_entity(self, entity):
        rec = {
            'email': entity.email,
            'password_hash': entity.password_hash,
            'first_name': entity.first_name,
            'last_name': entity.last_name,
            'birth_date': entity.birth_date.isoformat(),
        }

        if entity.id is not None:
            rec['_id'] = ObjectId(entity.id)

        return rec

    def get_by_email(self, email: str):
        result = self.collection.find_one({'email': email})
        return self.to_entity(result)
