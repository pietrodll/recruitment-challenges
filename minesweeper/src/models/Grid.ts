class Grid {

  protected grid: number[];
  protected size: [number, number];
  protected nBombs: number;

  constructor(size: [number, number], nBombs: number) {
    this.size = size;
    this.nBombs = nBombs;
    this.grid = this.createGrid(size);
    this.initGrid(nBombs);
  }

  /**
   * Creates a matrix of size n, with all the values equal to 0
   * @param n The size of the grid
   */
  private createGrid([n, m]: [number, number]): number[] {
    const grid = new Array(n * m).fill(0);
    return grid;
  }

  /**
   * Puts ``nBombs`` bombs in the grid. The value of the bombs in the grid is ``Infinity``.
   * @param nBombs The number of bombs
   */
  private putBombs(nBombs: number): Array<[number, number]> {
    const bombs: [number, number][] = [];
    const [n, m] = this.size;
    for (let k = 0; k < nBombs; k++) {

      // Choose random positions until there is not altready a bomb.
      // The probability of this to loop for a long time grows with nBombs,
      // but with the number we use it is very low.
      let i = Math.floor(Math.random() * n);
      let j = Math.floor(Math.random() * m);
      while (this.grid[i * m + j] === Infinity) {
        i = Math.floor(Math.random() * n);
        j = Math.floor(Math.random() * m);
      }
      this.grid[i * m + j] = Infinity;
      bombs.push([i, j]);
    }
    return bombs;
  }

  /**
   * Returns an array containing the coordinates of the neighbors of ``[i, j]``
   * @param i The line number
   * @param j The column number
   */
  public getNeighbors(i: number, j: number): Array<[number, number]> {
    const [n, m] = this.size;

    const neighbors: [number, number][] = [];
    if (i > 0) {
      neighbors.push([i-1, j]);
      if (j > 0) neighbors.push([i-1, j-1]);
      if (j < m - 1) neighbors.push([i-1, j+1]); 
    }
    if (i < n - 1) {
      neighbors.push([i+1, j]);
      if (j > 0) neighbors.push([i+1, j-1]);
      if (j < m - 1) neighbors.push([i+1, j+1]);
    }
    if (j > 0) neighbors.push([i, j-1]);
    if (j < m - 1) neighbors.push([i, j+1]);
    return neighbors;
  }

  /**
   * Initializes the grid, by putting ``nBombs`` bombs randomly and computing the values of the squares in the grid.
   * The utility of having given ``Infinity`` value to the bombs, is that 1 + ``Infinity`` = ``Infinity``
   * @param nBombs The number of bombs
   */
  private initGrid(nBombs: number): void {
    const m = this.size[1];
    const bombs = this.putBombs(nBombs);
    bombs.forEach(([i, j]) => {
      const neighbors = this.getNeighbors(i, j);
      neighbors.forEach(([x, y]) => {
        this.grid[x * m + y] += 1
      });
    });
  }
  
  /**
   * Get the value of the square [i, j] of the grid
   * @param i The line number
   * @param j The column number
   */
  public getValue(i: number, j: number): number {
    const [n, m] = this.size;
    if (i < 0 || i >= n || j < 0 || j >= m) {
      throw Error("Square out of grid");
    }
    return this.grid[i * m + j];
  }

  /**
   * Returns ``true`` if the square [i, j] of the grid is zero, ``false`` oftherwise
   * @param i The line number
   * @param j The column number
   */
  public isZero(i: number, j: number): boolean {
    const [n, m] = this.size;
    if (i < 0 || i >= n || j < 0 || j >= m) {
      throw Error("Square out of grid");
    }
    return this.grid[i * m + j] === 0;
  }

  /**
   * Returns ``true`` if the square [i, j] of the grid is a bomb, ``false`` oftherwise
   * @param i The line number
   * @param j The column number
   */
  public isBomb(i: number, j: number): boolean {
    const [n, m] = this.size;
    if (i < 0 || i >= n || j < 0 || j >= m) {
      throw Error("Square out of grid");
    }
    return this.grid[i * m + j] === Infinity;
  }

  /**
   * Returns the grid
   */
  public getFlatGrid(): number[] {
    return this.grid;
  }

  /**
   * Resets the grid
   * @param nBombs The number of bombs
   */
  public resetGrid(nBombs: number): void {
    this.grid = this.createGrid(this.size);
    this.initGrid(nBombs);
  }

}

export default Grid;
