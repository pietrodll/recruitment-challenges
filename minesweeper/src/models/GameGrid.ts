import Grid from './Grid';

export enum GameGridElement {
  Hidden,
  Flag,
  Displayed
}

class GameGrid extends Grid {

  private gameGrid: GameGridElement[];
  private nDiscovered: number;

  constructor(size: [number, number], nBombs: number) {
    super(size, nBombs);
    this.gameGrid = this.createGameGrid(size);
    this.nDiscovered = 0;
  }

  /**
   * Creates a game grid of size n, with all the values equal to ``GameGridElement.Hidden``
   * @param n The size of the grid
   */
  private createGameGrid([n, m]: [number, number]): GameGridElement[] {
    const grid = new Array(n * m).fill(GameGridElement.Hidden);
    return grid;
  }

  /**
   * Gets the game value of the square [i, j]
   * @param i The line number
   * @param j The column number
   */
  public getGameValue(i: number, j: number): GameGridElement {
    const [n, m] = this.size;
    if (i < 0 || i >= n || j < 0 || j >= m) {
      throw Error("Square out of grid");
    }
    return this.gameGrid[i * m + j];
  }

  /**
   * Gets the game grid
   */
  public getFlatGameGrid(): GameGridElement[] {
    return this.gameGrid;
  }

  /**
   * Discovers a square of the grid, and continues until all the zero neighbors are discivered too.
   * The functions uses a depth first search in the grid to discover all the neighbors of the zero squares.
   * @param i The line number
   * @param j The column number
   */
  public discover(i: number, j: number) {
    const [n, m] = this.size;

    let x = i;
    let y = j;
    const points: Array<[number, number]> = [];
    const visited: { [i: number]: boolean } = {};
    points.push([x, y]);
    while (points.length > 0) {
      // we need a default value to avoid a typescript error,
      // but as thanks to the while loop we know that the points array is not empty,
      // the default value will never be used, because points.pop() will never return undefined
      [x, y] = points.pop() || [0, 0];
      visited[x * m + y] = true;
      if (this.gameGrid[x * m + y] !== GameGridElement.Displayed) {
        this.nDiscovered += 1;
      }
      this.gameGrid[x * m + y] = GameGridElement.Displayed;
      if (this.isZero(x, y)) {
        this.getNeighbors(x, y).forEach(([xn, yn]) => {
          if (!visited[xn * m + yn]) {
            points.push([xn, yn]);
          }
        });
      }
    }
    const visitedSquares: Array<[number, number]> = [];
    Object.keys(visited).forEach(index => {
      const ind = parseInt(index, 10)
      const col = ind % m;
      const line = Math.floor(ind / m);
      visitedSquares.push([line, col]);
    })
    console.log('Discovered:', this.nDiscovered);
    return {
      discovered: this.nDiscovered,
      visited: visitedSquares
    };
  }

  /**
   * Puts a flag at the square [i, j]
   * @param i The line number
   * @param j The column number
   */
  public putFlag(i: number, j: number): [number, number] {
    const [n, m] = this.size;
    if (i < 0 || i >= n || j < 0 || j >= m) {
      throw Error("Square out of grid");
    }
    if (this.gameGrid[i * m + j] === GameGridElement.Hidden) {
      this.gameGrid[i * m + j] = GameGridElement.Flag;
    } else if (this.gameGrid[i * m + j] === GameGridElement.Flag) {
      this.gameGrid[i * m + j] === GameGridElement.Hidden;
    }
    return [i * m + j, this.gameGrid[i * m + j]];
  }

  /**
   * Resets the grid
   * @param nBombs The number of bombs to put in the grid
   */
  public resetGrid(nBombs: number): void {
    super.resetGrid(nBombs);
    this.nDiscovered = 0;
    this.gameGrid = this.createGameGrid(this.size);
  }

}

export default GameGrid;