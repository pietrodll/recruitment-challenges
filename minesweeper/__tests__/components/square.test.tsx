import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Square from '../../src/components/square';
import { GameGridElement } from '../../src/models/GameGrid';

describe('Square component', () => {
  it('renders correctly when displayed', () => {
    const tree = renderer
      .create(
        <Square
          num={1}
          gameState={GameGridElement.Displayed}
          onPress={() => {}}
          onLongPress={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when displaying an icon', () => {
    const tree = renderer
      .create(
        <Square
          num={1}
          gameState={GameGridElement.Flag}
          onPress={() => {}}
          onLongPress={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when hidden', () => {
    const tree = renderer
      .create(
        <Square
          num={1}
          gameState={GameGridElement.Hidden}
          onPress={() => {}}
          onLongPress={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
