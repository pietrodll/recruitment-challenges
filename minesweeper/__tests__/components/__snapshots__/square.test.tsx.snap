// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square component renders correctly when displayed 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#cccccc",
      "borderRadius": 3.5,
      "display": "flex",
      "height": 35,
      "justifyContent": "center",
      "margin": 2,
      "opacity": 1,
      "textAlign": "center",
      "width": 35,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`Square component renders correctly when displaying an icon 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2244ee",
      "borderRadius": 3.5,
      "display": "flex",
      "height": 35,
      "justifyContent": "center",
      "margin": 2,
      "opacity": 1,
      "textAlign": "center",
      "width": 35,
    }
  }
>
  <View>
    <SvgMock
      fill="#cccccc"
      height={25}
      width={25}
    />
  </View>
</View>
`;

exports[`Square component renders correctly when hidden 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2244ee",
      "borderRadius": 3.5,
      "display": "flex",
      "height": 35,
      "justifyContent": "center",
      "margin": 2,
      "opacity": 1,
      "textAlign": "center",
      "width": 35,
    }
  }
>
  <View>
    <Text />
  </View>
</View>
`;
